lib/ecto_enum.ex

```diff
@@ -57,18 +64,34 @@ defmodule EctoEnum do
     end
   end
 
+  defmodule UndeterminableStorageError do
+    defexception [:message]
+
+    def exception(kw) do
+      msg = "cannot determine storage type for #{inspect kw}"
+      %__MODULE__{message: msg}
+    end
+  end
+
   defmacro defenum(module, enum_kw) when is_list(enum_kw) do
     quote do
       kw = unquote(enum_kw) |> Macro.escape
 
+      storage = EctoEnum.storage(kw)
```

Comment:
The error will raised at compile-time still. That said, it would be cleaner outside of the quoted block.